### ISSUE TITLE
accessModeSufficient can be set to visual

### ISSUE THREAD

**Issue by CircularKen** [daisy/kb#39]
Hi DAISY 

I have a few questions after reading though the pages describing accessMode and accessModeSufficient on the KB.

The definition of textual contains this sentence - "If a publication contains images of text, the visual access mode is also set (or set instead, if the publication only contains images of text).” - which indicates that an accessMode value of _visual_ is sufficient to describe a publication that relies on the visual sense.

With accessModeSufficient defined as “a list of _single_ or combined access modes that are sufficient to understand all the intellectual content of a resource”, it seems that by entering a single entry of _visual_ the accessibility requirement can be declared. 

And it might follow that adding only a visual accessMode and accessModeSufficient to a complex fixed layout EPUB document, even one that doesn’t necessary have text as images, would mean that they are declaring that they are accessible but require the visual access mode? e.g.

```
<meta property="schema:accessMode">visual</meta>
<meta property="schema:accessModeSufficient">visual</meta>
```
Doing so passes ACE with no wcag2aa or EPUB violations.

Is this a loophole that is open to abuse and needs to be closed?
Or is it a legitimate use of schema.org accessibility metadata to identify the accessible qualities of a visual only publication?

**Comment by mattgarrish** [comment]
> would mean that they are declaring that they are accessible but require the visual access mode?

No, you're only declaring that you need to be sighted to fully consume the content. There are no other options.

The access mode is a statement about the content. In your case, there is only image/visual content in the book.

The sufficient access modes tell you the modalities/perceptions required to fully consume the content. So in this case, unless you can see the content, you can't read it.

Neither asserts conformance to WCAG or makes an explicit statement about the accessibility, although if you were to put in accessModeSufficient=textual for such a work you would be implying that none of the WCAG requirements related to images fails.

That said, you can assume that a purely visual work will fail WCAG. But passing Ace isn't proof you've passed WCAG, so the metadata alone isn't going to trigger a failure. It's only able to check some of the WCAG requirements. Human inspection is inevitable with accessibility testing.

And I don't believe Ace tries to validate misplaced discovery metadata like the a claim of a textual sufficient access mode when there is no text/alt text. It might be something to explore, though. /cc @danielweck @rdeltour @marisademeglio

**Comment by mattgarrish** [comment]
> Or is it a legitimate use of schema.org accessibility metadata to identify the accessible qualities of a visual only publication?

Sorry, forgot to answer this part, but it's yes.

Discovery metadata isn't only for accessible publications as not all publications are accessible to all users. Auditory works will also fail WCAG but are accessible to users who can't see the content, for example.

Including the metadata, regardless of accessibility, allows users to determine for themselves if they can read the work independent of conformance to WCAG or any other accessibility standard.

**Comment by danielweck** [comment]
> And I don't believe Ace tries to validate misplaced discovery metadata like a claim of a textual sufficient access mode when there is no text/alt text.

Correct.

**Comment by CircularKen** [comment]
Thanks for clarifying. As we work to improve the general accessibility of fixed layout EPUBs it's good to know how to correctly declare that, for some titles, the reader needs to be sighted to fully consume the content.

Rather than just declaring a binary visual yes or no, I wonder if there is scope for greater level of detail such adding information or warnings for colour blindness and partial sightedness for example?

**Comment by mattgarrish** [comment]
> I wonder if there is scope for greater level of detail such adding information or warnings for colour blindness and partial sightedness for example

I don't think this would work with access modes and sufficient access modes.

What you're sort of getting into is whether a publication represents an [optimization](https://www.w3.org/TR/epub-a11y-11/#sec-opt-pubs).

Colour blindness doesn't change the visual nature of the content, but knowing a publication is optimized for reading if you are colour blind is definitely important to know. It relies on there being a standard for assessing conformance, though, at least in terms of being able to report in the metadata.

WCAG 3's [functional categories](https://www.w3.org/TR/wcag-3.0/#functional-categories) also looks at delineating the requirements along the needs of users and breaks out of everything being lumped together (i.e., there will be more room for reporting specific failures for specific needs), but it's not coming any time soon.

In the absence of standardization, there's only the accessibility summary, where if you know your content is problematic for specific readers you could note the issues there.

**Comment by mattgarrish** [comment]
> I don't think this would work with access modes

I'll correct myself a bit on this. I took another look at the accessibility metadata wiki and it lists "colorDependent" as an access mode, but there's no description of the value. I went to the access for all page where it defines a "colour" value as:

>  Information is conveyed that requires the ability to perceive colour.

By that definition, though, it seems to be for monochromatic vision. I don't know if you can use it for red-green and other types of colour blindness, as it doesn't really give useful information about which colours are problematic.
